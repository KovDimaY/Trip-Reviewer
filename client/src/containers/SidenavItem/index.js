import React from 'react';
import { Link } from 'react-router-dom';
import FontAwesome from 'react-fontawesome';
import { connect } from 'react-redux';

const SidenavItem = ({ item, users }) => {
    const element = (
        <div className={item.type}>
            <Link to={item.link}>
                <FontAwesome name={item.icon}/>
                {item.text}
            </Link>
        </div>
    );

    const showItem = () => {
        if (users.login) {
            if (users.login.isAuth) {
                return !item.exclude 
                    ? element
                    : null;
            }
            return !item.restricted 
                ? element
                : null;
        } 
        return null;
    };
        
    return (
        <div>
            {showItem()}
        </div>
    );
};

function mapStateToProps(state) {
    return {
        users: state.users
    }
}

export default connect(mapStateToProps)(SidenavItem)