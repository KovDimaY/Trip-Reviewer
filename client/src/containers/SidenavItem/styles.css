.navItem {
    font-weight: 300;
    font-size: 12px;
    color:#bababa;
    padding:10px;
    border-top: 1px solid #404040;
    display: block;
}

.navItem a {
    color:inherit;
    text-decoration: none;
}

.navItem span {
    color:white;
    margin-right:10px;
}