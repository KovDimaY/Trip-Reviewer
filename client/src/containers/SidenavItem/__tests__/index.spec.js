import React from 'react';
import { create } from 'react-test-renderer';
 
import SidenavItem from './../../SidenavItem';
import Root from './../../../hoc/root';

const mockComponent = props => {
    return (
        <Root>
            <SidenavItem {...props} />
        </Root>
    );
};
 
describe('<SidenavItem />', () => {
    it('should render component', () => {
        const props = {
            users: {},
            item: {
                type: 'type',
                link: 'link',
                icon: 'icon',
                text: 'text'
            }
        };
        const tree = create(mockComponent(props)).toJSON();
        
        expect(tree).toMatchSnapshot();
    });
});