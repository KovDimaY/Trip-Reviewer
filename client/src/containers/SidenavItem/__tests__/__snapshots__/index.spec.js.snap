// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidenavItem /> should render component when no authenticated 1`] = `
<div>
  <div
    className="type"
  >
    <Link
      to="link"
    >
      <span
        aria-hidden={true}
        className="fa fa-icon"
      />
      text
    </Link>
  </div>
</div>
`;

exports[`<SidenavItem /> should render component when no authenticated and restricted 1`] = `<div />`;

exports[`<SidenavItem /> should render component with authentication 1`] = `
<div>
  <div
    className="type"
  >
    <Link
      to="link"
    >
      <span
        aria-hidden={true}
        className="fa fa-icon"
      />
      text
    </Link>
  </div>
</div>
`;

exports[`<SidenavItem /> should render component with authentication and excluded 1`] = `<div />`;

exports[`<SidenavItem /> should render component with no login 1`] = `<div />`;
